#!/usr/bin/env python
PACKAGE = "kalman_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

KALMAN_FILTER_FLAG = 0
INPUT_SIGMA = 1
BIAS_SIGMA = 2
MEASURE_SIGMA = 3

gen = ParameterGenerator()

gen.add("kalmanFilterFlag", bool_t, KALMAN_FILTER_FLAG, "Kalman Filter Flag", False)
gen.add("input1Sigma", double_t, INPUT_SIGMA, "Input Sigma", 0.04, 0, 1)
gen.add("input2Sigma", double_t, BIAS_SIGMA, "Bias Sigma", 0.0005, 0, 1)
gen.add("measureSigma", double_t, MEASURE_SIGMA, "Measure Sigma", 0.04, 0, 1)

exit(gen.generate(PACKAGE, "kalman_filter", "KalmanFilterPosVelAccBias"))


